atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/SnapshotTransaction.java

```diff
@@ -1305,70 +1312,113 @@ private void commitWrites(TransactionService transactionService) {
-            }
+                return ImmutableLogTemplate.builder().format(
+                        "Committed {} bytes with locks, start ts {}, commit ts {}, "
+                                + "acquiring locks took {} us, checking for conflicts took {} us, "
+                                + "writing to the sweep queue took {} us, "
+                                + "writing data took {} us, "
+                                + "getting the commit timestamp took {} us, punch took {} us, "
+                                + "serializable r/w conflict check took {} us, putCommitTs took {} us, "
+                                + "pre-commit lock checks took {} us, user pre-commit conditions took {} us, "
+                                + "total time spent committing writes was {} us, "
+                                + "total time since tx creation {} us, tables: {}, {}.")
+                        .arguments(
+                                SafeArg.of("numBytes", byteCount.get()),
+                                SafeArg.of("startTs", getStartTimestamp()),
+                                SafeArg.of("commitTs", commitTimestamp),
+                                SafeArg.of("microsForLocks", microsForLocks),
+                                SafeArg.of("microsCheckForConflicts", microsCheckingForConflicts),
+                                SafeArg.of("microsWritingToTargetedSweepQueue",
+                                        microsWritingToTargetedSweepQueue),
+                                SafeArg.of("microsForWrites", microsForWrites),
+                                SafeArg.of("microsForGetCommitTs", microsForGetCommitTs),
+                                SafeArg.of("microsForPunch", microsForPunch),
+                                SafeArg.of("microsForReadWriteConflictCheck", microsForReadWriteConflictCheck),
+                                SafeArg.of("microsForPutCommitTs", microsForPutCommitTs),
+                                SafeArg.of("microsForPreCommitLockCheck", microsForPreCommitLockCheck),
+                                SafeArg.of("microsForUserPreCommitCondition", microsForUserPreCommitCondition),
+                                SafeArg.of("microsForCommitStage", microsForCommitStage),
+                                SafeArg.of("microsSinceCreation", microsSinceCreation),
+                                tableRefs.safeTableRefs(),
+                                tableRefs.unsafeTableRefs())
+                        .build();
+                    });
         } finally {
             timelockService.unlock(ImmutableSet.of(commitLocksToken));
         }
     }
 
+    private void updateNonPutOverheadMetrics(long microsWritingToTargetedSweepQueue, long microsForWrites,
+            long microsForCommitStage) {
+        long nonPutOverhead = microsForCommitStage - microsForWrites - microsWritingToTargetedSweepQueue;
+        getTimer("nonPutOverhead").update(nonPutOverhead, TimeUnit.MILLISECONDS);
```

Comment:
I think this is in micros as all of the inputs to the calculation are in micros

Reply:
Ooh, good catch! Will fix. Yep, a better fix is to switch everything out to Duration but agree that in this instance this fix will do